### PR TITLE
complete Onboarding Day-2/Part-1 : Feature/itunes

### PR DESCRIPTION
**Pull request by apurv-wednesday**
### Ticket Link

NA

### Related Links

https://docs.google.com/document/d/1vRY80RhKnw6SDxd34RKN9FdRn4GKsJ2PL4TFeBAA57U/edit#heading=h.ao8blx2gmp1d

### Description
*  Add basic Skelton for the iTunes app.  
* Add services, reducer, saga & selectors.
* create tests for all components.

### Steps to Reproduce / Test

to start the app
```bash
yarn start
```

to test
```bash
yarn test
```
---

### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's
Screenshot 2024-03-19 at 4 55 55 PM
Screenshot 2024-03-19 at 4 57 31 PM

### Live Link

---

- http://wednesday-react-template.s3-website.ap-south-1.amazonaws.com/<BRANCH_NAME>

### CODE REVIEW

File: app/services/tests/itunesAPi.test.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+//     const searchTerm = 'coldplay';
+//     const response = await getSongs(searchTerm);
+//     console.log(response)
+//     console.log(response.data)
+
+    
+//     expect(response.status).toBe(200);
+//     expect(response.data.results.length).toBeGreaterThan(0);
+//   });
+// });
```
Comment (praveenkumar1798) [inline app/services/tests/itunesAPi.test.js:32]: Why is this commented ?

File: app/containers/Itunes/index.js [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+      dispatchSongsApi(songName);
+    }
+  }, [dispatchSongsApi, dispatchClearSongsData, songName, songsData]);
+
+  const searchSongs = (sName) => {
+    dispatchSongsApi(sName);
+  };
+
+  const handleOnChange = (sName) => {
+    console.log(sName); // eslint-disable-line no-console
```
Comment (praveenkumar1798) [inline app/containers/Itunes/index.js]: Do we need this ?

File: app/services/tests/itunesAPi.test.js (hunk 1 above)
Comment (apurv-wednesday) [reply app/services/tests/itunesAPi.test.js:32]: fixed!

File: app/containers/Itunes/index.js (hunk 2 above)
Comment (apurv-wednesday) [reply app/containers/Itunes/index.js]: fixed!

**Review by praveenkumar1798 — APPROVED** [review]

**Pull request merged**